core-contracts/contracts/hyperchains.clar

```
@@ -56,11 +56,17 @@
    ))
 
 ;; Helper function: determines whether the commit-block operation can be carried out
-(define-private (can-commit-block? (commit-block-height uint))
+(define-private (can-commit-block? (commit-block-height uint)  (target-block (buff 32)))
     (begin
         ;; check no block has been committed at this height
         (asserts! (is-none (map-get? block-commits commit-block-height)) (err ERR_BLOCK_ALREADY_COMMITTED))
 
+        ;; check that `target-block` matches the burn chain tip
+        (asserts! (is-eq 
+            target-block 
+            (unwrap! (get-block-info? id-header-hash (- block-height u1)) (err ERR_BLOCK_ALREADY_COMMITTED)) )
```

I would use a different error here, since this error should be impossible - get-block-info? should only fail for an invalid block-height. Maybe something like ERR_UNEXPECTED_PANIC...

---

I used `ERR_CALLED_TOO_EARLY`.. it's unexpected at a system level that we'd have this. But, as a module, this wouldn't be an internal error to the module. wdyt?